src/parser.rs

```diff
@@ -605,16 +611,20 @@ impl<'tokens, 'src> Parser<'tokens, 'src> {
 
     let body = self.parse_body()?;
 
-    Ok(Recipe {
+    let recipe = Recipe {
       private: name.lexeme().starts_with('_'),
       shebang: body.first().map(Line::is_shebang).unwrap_or(false),
       parameters: positional.into_iter().chain(variadic).collect(),
-      doc,
+      doc: doc.map(|x| x[1..].trim_start()),
```

Comment:
This is problematic. The logic of adding comment looks like:
- if comment is found and next token is not `Identifier` then add comment
- otherwise next `Identifier` could be:
  - if `Recipe` then add comment to a docstring
  - otherwise insert comment to a tree before that non-recipe identifier

The point of `&mut` is to signal `Recipe` to `parse_justfile` by setting `*doc = None`. Alternatives to this are:
- find a way to tell whether next identifier is recipe (which seems difficult because lines 317-355 are all about it)
- set `doc = None` in five possible places where `parse_recipe` is called

Current solution seemed a lesser evil. If there's an elegant one-line call to determine whether next identifier is recipe than it would be way better than this.

Reply:
Thanks for the explanation. I agree, the current solution sounds like the best option. The logic of determining if the next identifier is quite complicated.